### PR TITLE
api: add undocumented /grpc endpoint available since API 1.40

### PR DESCRIPTION
**Pull request by tiborvass**
Signed-off-by: Tibor Vass <tibor@docker.com>

This allows to use buildx with docker driver on swarm.

### CODE REVIEW

**Comment by tiborvass** [comment]
Ping @tonistiigi @dperny @dhiltgen

File: api/handlers.go [hunk 1]
```diff
@@ -1670,3 +1670,23 @@ func stripNodeNamesFromNetworkingConfig(n network.NetworkingConfig, nodeNames []
 	n.EndpointsConfig = endpointsConfig
 	return n
 }
+
+// postGRPC randomly chooses a node on the cluster and hijacks that connection.
+func postGRPC(c *context, w http.ResponseWriter, r *http.Request) {
+	engine, err := c.cluster.RANDOMENGINE()
```
Comment (dhiltgen) [inline api/handlers.go]: Not ideal, but seems acceptable for the time being.

**Comment by tiborvass** [comment]
@dperny is the CI failure unrelated?

**Comment by dperny** [comment]
ci failure is almost certainly unrelated, but i don't know anything about the ci system here.

**Comment by dperny** [comment]
looks fine.

**Comment by thaJeztah** [comment]
Upstream changes in the engine; https://github.com/moby/moby/pull/38990

**Pull request closed without merging**